### DISCUSSION TITLE
How to save Best Precision while training

### DISCUSSION THREAD

**Discussion by sarar1997** [ultralytics/ultralytics#7778] [Q&A]
Hi, How we can set the YOLO saves the best precision model ?  lets say I am not interested in mAP50 or mAp50-95, How can I save the best precision model end of the training? 
Thank you  @glenn-jocher

**Comment by glenn-jocher** [comment]
@sarar1997 hello! To save the model with the best precision during training, you can modify the validation code to track precision and save the model when it achieves the highest precision score. However, this feature is not directly available in the current YOLOv8 repository, so it would require custom code changes.

If you're comfortable with modifying the source code, you can track the precision metric by editing the validation function in the training script. You would need to calculate precision during validation and then update the checkpoint saving logic to save the model with the best precision instead of the best mAP.

If you need further assistance with the code changes or have any other questions, feel free to ask! 😊👨‍💻
